Zend/tests/attributes/deprecated_001.phpt

```diff
@@ -0,0 +1,47 @@
+--TEST--
+#[Deprecated] attribute
+--FILE--
+<?php
+
+error_reporting(E_ALL | E_DEPRECATED);
+ini_set('display_errors', true);
+
+#[Deprecated]
+function test() {
+}
+
+#[Deprecated("use test() instead")]
+function test2() {
+}
+
+class Clazz {
+    #[Deprecated]
+    function test() {
+    }
+
+    #[Deprecated("use test() instead")]
+    function test2() {
+    }
+}
+
+test();
+test2();
+call_user_func("test");
+
+$cls = new Clazz();
+$cls->test();
+$cls->test2();
+
+call_user_func([$cls, "test"]);
+--EXPECTF--
+Deprecated: Function test() is deprecated in %s
+
+Deprecated: Function test2() is deprecated, use test() instead in %s
```

Comment:
Something like `is deprecated. Reason: %s` would be my preference for what may end up being seen in practice, such as `Deprecated('inefficient')`, 'will remove', 'Sentence 1. Sentence 2.', etc., but there's probably an argument that people should match deprecation messages to whatever format is expected by PHP and I'm not great at writing error messages.

Should this avoid the extra formatting for the empty string `#[Deprecated('')]`?